exporter/stats/stackdriver/stackdriver.go

```diff
@@ -101,9 +102,41 @@ type Options struct {
 	Resource *monitoredrespb.MonitoredResource
 }
 
+// Enforces the singleton on NewExporter per projectID per process
+// lest there will be races with Stackdriver.
+var seenProjectIDs struct {
+	m map[string]bool
+	sync.Mutex
+}
+
+func init() {
+	seenProjectIDs.m = make(map[string]bool)
+}
+
+var (
+	errBlankProjectID    = errors.New("expecting a non-blank ProjectID")
+	errSingletonExporter = errors.New("only one exporter can be created per unique ProjectID per process")
+)
+
 // NewExporter returns an exporter that uploads stats data to Stackdriver Monitoring.
-// Only one Stackdriver exporter should be created per process.
+// Only one Stackdriver exporter should be created per ProjectID per process, any subsequent
+// invocations of NewExporter with the same ProjectID will return an error.
 func NewExporter(o Options) (*Exporter, error) {
+	if strings.TrimSpace(o.ProjectID) == "" {
```

Comment:
seenProjectIDs.Lock()
defer seenProjectIDs.Unlock()

We dont care holding this mutex in the lifetime of NewExporter. This function will be called very few times.

Reply:
True true.